website/docs/r/workspaces_directory.html.markdown

```diff
@@ -81,6 +109,8 @@ resource "aws_workspaces_directory" "example" {
     enable_maintenance_mode             = true
     user_enabled_as_local_administrator = true
   }
+
+  depends_on = [aws_iam_role.workspaces_default]
```

Comment:
Typically, services depend on the `aws_iam_role_policy_attachment` resources. In this case does it just check for the existence of the role itself?

Reply:
Yes. This is quite strange for me too – AWS WorkSpaces service requires the exact `workspaces_DefaultRole` IAM role name along with a particular set of policies. I'm happy with the case when service allows me to provide my custom role, like API Gateway CloudWatch IAM role.

Reply:
Reverse, or even just move the `aws_workspaces_directory` to the top

Reply:
If it needs the policies as well, we should make the dependency on the `aws_iam_role_policy_attachment` resources. If all it needs is the IAM role, and the policies aren't strictly needed at creation time of the WorkSpaces directory, then the dependency can be on the `aws_iam_role` itself. Otherwise, we can get into race conditons where the configuration works sometimes and not others. Examples should be copy-pastable as much as possible.

Reply:
Good point and it makes sense to me. Pollicies are required, so I'll put a dependency on the attachment to rid of race condtions.